docs/spanner/snapshot-usage.rst

```diff
@@ -64,9 +64,24 @@ fails if the result set is too large,
 
 .. note::
 
-   If streaming a chunk fails due to a "resumable" error,
-   :meth:`Session.read` retries the ``StreamingRead`` API reqeust,
-   passing the ``resume_token`` from the last partial result streamed.
+   If streaming a chunk raises an exception, the application can
+   retry the ``read``, passing the ``resume_token`` from ``StreamingResultSet``
+   which raised the error.  E.g.:
+
+   .. code:: python
+
+      result = snapshot.read(table, columns, keys)
+      while True:
+          try:
+              for row in result.rows:
+                  print row
+          except Exception:
+               result = snapshot.read(
+                  table, columns, keys, resume_token=result.resume_token)
```

Comment:
This is not completely correct. As I had mentioned resume token might not be present in every single PRS. In that case this field would be empty which mean this code would end reading everything again and thus seeing duplicate rows. Or the other way round where the resume token was there but we did not yet have a complete row. For eg let us say a single row was chunked into 3 chunks PRS1, PRS2, PRS3. PRS1 was received which had a resume token. Stream broke after that. Now if the user resumes we will only fetch PRS2 and PRS3 but PRS1 has been lost since it was in a different iterator. 

 More I think about this, it seems to be really tricky to document exactly what the user needs to do in the case of a broken stream. Its better to just fix this in the code.